tools/monitoring/landsat_gap_report.py

```diff
@@ -1 +1,299 @@
+                str(USGS_S3_BUCKET_PATH / path)
+                for path in source_paths.difference(dest_paths)
+            ]
+
+            # Keys that are orphan, they are in the bucket but not found in the files
+            log.info("Filtering orphan scenes")
+            orphaned_scenes = [
+                str(URL(f"s3://{bucket_name}") / path)
+                for path in dest_paths.difference(source_paths)
+            ]
+
+            log.info(f"missing_scenes 10 first keys {list(missing_scenes)[0:10]}")
+            log.info(f"orphaned_scenes 10 first keys {list(orphaned_scenes)[0:10]}")
+
+        landsat_s3 = s3_client(region_name="af-south-1")
+
+        if len(missing_scenes) > 0:
+            output_filename = (
+                f"{satellite_name}_{date_string}.txt.gz"
+                if not update_stac
+                else f"{satellite_name}_{date_string}_update.txt.gz"
+            )
+
+            log.info(
+                f"File will be saved in {URL(landsat_status_report_path) / output_filename}"
+            )
+            s3_dump(
+                data=gzip.compress(str.encode("\n".join(missing_scenes))),
+                url=str(URL(landsat_status_report_path) / output_filename),
+                s3=landsat_s3,
+                ContentType="application/gzip",
+            )
+
+            log.info(f"Number of missing scenes: {len(missing_scenes)}")
+
+        if len(orphaned_scenes) > 0:
+            orphan_output_filename = f"{satellite_name}_{date_string}_orphaned.txt.gz"
+            s3_dump(
+                data=gzip.compress(str.encode("\n".join(orphaned_scenes))),
+                url=str(URL(landsat_status_report_path) / orphan_output_filename),
+                s3=landsat_s3,
+                ContentType="application/gzip",
+            )
+
+            log.info(f"Number of orphaned scenes: {len(orphaned_scenes)}")
+
+        message = dedent(
+            f"*{satellite_name.upper()} GAP REPORT*\n "
+            f"Environment: {environment}\n "
+            f"Missing Scenes: {len(missing_scenes)}\n"
+            f"Orphan Scenes: {len(orphaned_scenes)}\n"
+            f"Missing Scenes report Saved: {str(URL(landsat_status_report_path) / output_filename)}\n"
+            f"Orphan Scenes report Saved: {str(URL(landsat_status_report_path) / orphan_output_filename)}\n"
+        )
+
+        log.info(message)
+
+        if len(missing_scenes) > 0 or len(orphaned_scenes) > 0:
+
+            log.info(
+                f"File {file_name} processed and sent in {time_process(start=start_timer)}"
+            )
+
+            if notification_url is not None:
+                send_slack_notification(
+                    notification_url, f"{satellite_name} Gap Report", message
+                )
+
+            if not update_stac and (
+                len(missing_scenes) > 200 or len(orphaned_scenes) > 200
+            ):
+                raise Exception(f"More than 200 scenes were found \n {message}")
```

Comment:
I think sending the notification should happen only if tehre are > 200 scenes otherwise we'll get notified all the time.